luxtronik/__main__.py

```diff
@@ -0,0 +1,41 @@
+"""Luxtronik CLI."""
+import argparse
+import sys
+from luxtronik.discover import discover as _discover
+from luxtronik.scripts.dump import dump  # pylint: disable=unused-import # noqa: F401
+
+
+def discover():
+    """Output available heat pump controllers nicely."""
+    results = _discover()
+    print(f"{len(results)} heatpump(s) reported back")
+    for count, result in enumerate(results):
+        print(f"Heat pump #{count} -> IP address: {result[0]} port: {result[1]}")
+
+
+def main() -> int:
+    """Entry point to parse CLI arguments."""
+    parser = argparse.ArgumentParser(
+        description="CLI for Luxtronik controllers",
+        usage="""luxtronik <command> [<args>]
+        The supported commands are:
+        dump       Dump all available data from the Luxtronik controller 
+        discover   Discover Luxtronik controllers on the network (via magic packet) and output results
+        """,
+    )
+    parser.add_argument("command", help="Subcommand to run")
+    # parse_args defaults to [1:] for args, but you need to
+    # exclude the rest of the args too, or validation will fail
+    args = parser.parse_args(sys.argv[1:2])
+    commands = {
+        "dump": dump,
+        "discover": discover,
```

Comment:
```suggestion
        "dump": dump,
        "discover": discover,
```

I would suggest to use the fully qualified name here. This will make it more explicit that this is a reference to a function.

Reply:
I think I had something like this in mind back in March:

```suggestion
        "dump": luxtronik.dump,
        "discover": luxtronik.discover,
```

Reply:
OK, lets merge this one and if we are not happy we can always change it later on!